### DISCUSSION TITLE
Using a custom *map and changing behavior of lock LEDs

### DISCUSSION THREAD

**Discussion by RaoulDuke-Esq** [tmk/tmk_keyboard#699] [Q&A]
Hi Hasu,

I'm loving ibmpc_usb and have been using the online GUI but I'm ready to take it to the next level.  I tried creating a keymap_raoulduke-esq.c in my local fork and using make -f Makefile.atmega32u4 [KEYMAP=raoulduke-esq] but it said it could not find the keymap.  How do I successfully build the firmware with a custom keymap when the default behavior is to use a unimap?

Once I've got my custom keymap working, I wanted to make my ibmpc devices use the lock lights as layer status indicators (i.e. Layer 0 active = NumLock light on; layer 1 active = CapsLock on, etc.).  I use a Mac so numlock doesn't have a role, I don't use Capslock, and there's no scroll lock either so I figured layer lights would let me see these gorgeous green LEDs.  Could you give me an example of how to use the a hook with my converter to make this happen?

**Comment by tmk** [comment]
you need `unimap_yourname.c` probably.

**Reply by RaoulDuke-Esq** [reply]
Thanks!  I'll try that and mark as the answer if successful.

Any ideas about the LEDs?

**Comment by tmk** [comment]
You can use `hook_layer_change` to know layer state and change indicators.
https://github.com/tmk/tmk_keyboard/blob/master/tmk_core/common/hook.h#L78

Also you may need to override default behaviour using `hook_keyboard_leds_change` or `led_set`.

Refer to `led_set` in ibmpc_usb.c for how to handle the indicators.

There is some discussion about this around this comment.
https://geekhack.org/index.php?topic=41989.msg2999753#msg2999753
